modules/core/src/main/scala/com.snowplowanalytics.iglu/client/validator/CirceValidator.scala

```diff
@@ -102,4 +106,112 @@ object CirceValidator extends Validator[Json] {
 
   private def fromValidationMessage(m: ValidationMessage): ValidatorReport =
     ValidatorReport(m.getMessage, m.getPath.some, m.getArguments.toList, m.getType.some)
+
+  private def evaluateSchema(
+    schemaAsNode: JsonNode
+  ): Either[ValidatorError.InvalidSchema, JsonSchema] = {
+    Either
+      .catchNonFatal(
+        IgluMetaschemaFactory
+          .getSchema(schemaAsNode, SchemaValidatorsConfig)
+      )
+      .leftMap(ValidatorError.schemaIssue)
+  }
+
+  private def validateSchemaAgainstV4(schema: JsonNode): List[ValidatorError.SchemaIssue] = {
+    V4Schema
+      .validate(schema)
+      .asScala
+      .toList
+      .map(m => ValidatorError.SchemaIssue(m.getPath, m.getMessage))
+  }
+
+  private[client] object WithCaching {
+    type SchemaEvaluationKey         = (SchemaKey, TTL)
```

Comment:
Should be `StorageTime` instead of `TTL`? IIRC on 1.x there was some comment with the link to the issue, could we put something similar here as well?

Reply:
Ah yes, you are right - that should be `StorageTime`. Fixed now, and reintroduced the comment.